core/src/main/java/org/infinispan/container/impl/AbstractInternalDataContainer.java

```diff
@@ -138,11 +148,17 @@ public void put(int segment, K k, V v, Metadata metadata) {
          if (trace)
             log.tracef("Store %s in container", copy);
 
-         entries.compute(copy.getKey(), (key, entry) -> {
-            computeEntryWritten(key, copy);
-            activator.onUpdate(key, entry == null);
-            return copy;
-         });
+         // If passivation is enabled, we currently have to do this while holding the CHM lock to ensure atomicity.
+         // This should hopefully be fixed at some point in the future
+         if (hasPassivation) {
+            entries.compute(k, (key, entry) -> {
+               computeEntryWritten(key, copy);
```

Comment:
Not sure I understand what you want me to rename. The `maps` is the segment based maps and the `entries` is the actual map of all entries for eviction.

The reason that `computeEntryWritten` exists is because of this https://github.com/ben-manes/caffeine/issues/245